float64 low
float64 high
